### PR TITLE
Add material properties for metallic uranium

### PR DESCRIPTION
**Pull request by keckler**
## Description

The `Uranium` material class (metallic uranium) had very little substance. I happen to need this material for my work, so I added the following material properties:
* density
* heat capacity
* linear expansion
* linear expansion percent

I added references for the properties that were added, all from openly-available, reputable sources.

This should be merged after #1242 to reduce merge conflicts.

The test added here will need to be updated if #1077 is ever implemented.

---

## Checklist

- [x] This PR has only one purpose or idea.
- [x] Tests have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [x] The documentation is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

File: armi/materials/uranium.py [hunk 1]
```diff
@@ -20,35 +20,280 @@
...
+        17.88,
+        17.76,
+        17.53,
+        17.52,
+        16.95,
+        16.84,
+        16.71,
+    ]  # g/cc
+
+    _linearExpansionPercent = [
```
Comment (john-science) [inline armi/materials/uranium.py:153]: 1. Can we get references here?
2. Are these references fully open-sourcable?

File: armi/materials/uranium.py [hunk 2]
```diff
@@ -20,35 +20,280 @@
...
-from armi.materials.material import Material
+from numpy import interp
+
+from armi.materials.material import FuelMaterial
+from armi.nucDirectory import nuclideBases as nb
 from armi.utils.units import getTk
 
 
-class Uranium(Material):
+class Uranium(FuelMaterial):
```
Comment (john-science) [inline armi/materials/uranium.py:30]: I mean, obviously, stated in this one line, this change seems reasonable.

But.... can we run downstream integration tests and simulations to show that this doesn't drastically change our baselines?  Or that we can update our baselines to match.

The downstream changes are my only high-level concern.

File: armi/materials/uranium.py (hunk 1 above)
Comment (keckler) [reply armi/materials/uranium.py:153]: The references were added to the class `references` dict:
https://github.com/terrapower/armi/pull/1247/files#diff-892fc46b16c9e9253d76ae4229b186fcde69674d805f3e3db5ca27ced4b32c4fR205-R216

Let me know if that's not sufficient.

All of them come from open documents, from Argonne.

File: armi/materials/uranium.py (hunk 2 above)
Comment (john-science) [reply armi/materials/uranium.py:30]: Thanks!

File: armi/materials/uranium.py (hunk 1 above)
Comment (john-science) [reply armi/materials/uranium.py:153]: Oh, no, that's exactly what I wanted.  I am trying out the new GH code preview and I guess the slight change in UI got me.

Thanks!

**Review by john-science — APPROVED** [review]

**Pull request merged**